nilearn/datasets/func.py

```diff
@@ -1581,93 +1588,119 @@ def fetch_cobre(n_subjects=10, data_dir=None, url=None, verbose=1):
 
         - 'func': string list
             Paths to Nifti images.
-        - 'mat_files': string list
-            Paths to .mat files of each subject.
-        - 'phenotypic': ndarray
+        - 'confounds': string list
+            Paths to .tsv files of each subject, confounds.
+        - 'phenotypic': numpy.recarray
             Contains data of clinical variables, sex, age, FD.
         - 'description': data description of the release and references.
+        - 'desc_con': str
+            description of the confounds variables
+        - 'desc_phenotypic': str
+            description of the phenotypic variables.
 
     Notes
     -----
-    More information about datasets structure, See:
-    https://figshare.com/articles/COBRE_preprocessed_with_NIAK_0_12_4/1160600
+    See `more information about datasets structure
+    <https://figshare.com/articles/COBRE_preprocessed_with_NIAK_0_17_-_lightweight_release/4197885>`_
     """
+
     if url is None:
         # Here we use the file that provides URL for all others
-        url = "https://figshare.com/api/articles/1160600/15/files"
-
+        url = 'https://api.figshare.com/v2/articles/4197885'
     dataset_name = 'cobre'
     data_dir = _get_dataset_dir(dataset_name, data_dir=data_dir,
                                 verbose=verbose)
     fdescr = _get_dataset_descr(dataset_name)
 
     # First, fetch the file that references all individual URLs
-    files = _fetch_files(data_dir,
-                         [("files", url + "?offset=0&limit=300", {})],
+    files = _fetch_files(data_dir, [("4197885", url, {})],
                          verbose=verbose)[0]
+
     files = json.load(open(files, 'r'))
+    files = files['files']
     # Index files by name
     files_ = {}
     for f in files:
         files_[f['name']] = f
     files = files_
 
     # Fetch the phenotypic file and load it
-    csv_name = 'cobre_model_group.csv'
-    csv_file = _fetch_files(
-        data_dir, [(csv_name, files[csv_name]['downloadUrl'],
-                    {'md5': files[csv_name].get('md5', None),
-                     'move': csv_name})],
+    csv_name_gz = 'phenotypic_data.tsv.gz'
+    csv_name = os.path.splitext(csv_name_gz)[0]
+    csv_file_phen = _fetch_files(
+        data_dir, [(csv_name, files[csv_name_gz]['download_url'],
+                    {'md5': files[csv_name_gz].get('md5', None),
+                     'move': csv_name_gz,
+                     'uncompress': True})],
         verbose=verbose)[0]
 
     # Load file in filename to numpy arrays
-    names = ['id', 'sz', 'age', 'sex', 'fd']
-    csv_array = np.recfromcsv(csv_file, names=names, skip_header=True)
-    # Change dtype of id and condition column
-    csv_array = csv_array.astype(
-        [('id', '|U17'),
-         ('sz', '<i8'),
-         ('age', '<f8'),
-         ('sex', '<i8'),
-         ('fd', '<f8')])
```

Comment:
Its not a CSV table anymore, its TSV ;)
In case of this type conversion I had an encode problem in Pyhhon 3
Looks like it identifies correctly the type

```
rec.array([(40061, 18, 'Male', 'Right', 'Control', 'None', 133, 0.25512, 0.22657),
 (40090, 18, 'Female', 'Right', 'Control', 'None', 150, 0.16963, 0.16963),
 (40046, 18, 'Male', 'Left', 'Patient', '295.70 depressed type', 76, 0.37504, 0.30042)], 
 dtype=[('id', '<i8'), ('current_age', '<i8'), ('gender', 'S6'), ('handedness', 'S5'), ('subject_type', 'S7'), ('diagnosis', 'S21'), ('frames_ok', '<i8'), ('fd', '<f8'), ('fd_scrubbed', '<f8')])
```